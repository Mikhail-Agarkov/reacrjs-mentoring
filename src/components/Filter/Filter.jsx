//@flow
import React from 'react';
import styles from './filter.module.scss'

type Props = {
    title: string,
    selected: string,
    buttons: Array<Button>
}

type Button = {
    text: string,
    value: string,
    key: number,
    handler: Function
}

const Filter = (props: Props) => (
    <div className={styles.content}>
        <span className={styles.text}>{props.title}</span>
        <div className={styles.buttonGroup}>
            {props.buttons.map(button =>
                <button className={styles.button} onClick={button.handler}
                        data-selected={props.selected === button.value} key={button.key}>
                    {button.text}
                </button>)}
        </div>
    </div>
);

export default Filter;
