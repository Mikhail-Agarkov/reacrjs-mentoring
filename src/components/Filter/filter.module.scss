.content {
  display: flex;
  margin: 15px 0;
}

.buttonGroup {
  margin-left: 20px;
}

.button {
  border: none;
  outline: none;
  padding: 0 20px;
  background-color: rgba(66, 66, 66, 0.68);
  color: #FFFFFF;
  font-family: Montserrat;
  font-size: 15px;
  line-height: 40px;
  text-transform: uppercase;
  cursor: pointer;

  &:first-child {
    border-radius: 4px 0 0 4px;
  }

  &:last-child {
    border-radius: 0 4px 4px 0;
  }

  &:hover {
    background-color: #dd4957;
  }

  &:active {
    background-color: #c4414d;
  }
}

.text {
  color: #FFFFFF;
  font-family: Montserrat;
  font-size: 15px;
  line-height: 40px;
  text-transform: uppercase;
}