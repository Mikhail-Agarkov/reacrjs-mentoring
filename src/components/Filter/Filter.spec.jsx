import React from 'react';
import {shallow} from 'enzyme';

import Filter from './Filter';

describe('Filter', () => {
    it('should render Filter correctly', () => {
        const component = shallow(<Filter title='Sort by' buttons={
            [
                {
                    'handler': 'someFunc()',
                    'text': 'genre',
                    'key': 1
                },
                {
                    'handler': 'someFunc()',
                    'text': 'title',
                    'key': 2
                }
            ]
        }/>);
        expect(component).toMatchSnapshot();
    })
});