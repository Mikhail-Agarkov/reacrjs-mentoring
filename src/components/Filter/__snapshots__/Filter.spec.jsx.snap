// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render Filter correctly 1`] = `
<div
  className="content"
>
  <span
    className="text"
  >
    Sort by
  </span>
  <div
    className="buttonGroup"
  >
    <button
      className="button"
      data-selected={true}
      key="1"
      onClick="someFunc()"
    >
      genre
    </button>
    <button
      className="button"
      data-selected={true}
      key="2"
      onClick="someFunc()"
    >
      title
    </button>
  </div>
</div>
`;
